contrib/drivers/lvm/lvm.go

```diff
@@ -110,6 +112,24 @@ func (d *Driver) CreateVolume(opt *pb.CreateVolumeOpts) (*model.VolumeSpec, erro
 		}
 	}
 
+	// Copy snapshot to volume
+	var snap = opt.GetSnapshotId()
```

Comment:
I think the operation of create volume and create volume from snapshot can not be executed at the same time, so you should add a switch flag to identify which operation should be performed.

Maybe I didn't understand, please explain what is `dd` command used for.

Reply:
They will both be executed when creating from snapshots. lvcreate at first creates an lv, and then dd copies data from source snapshot lv to the new lv.  LVM has no command to create a thick lv from a snapshot directly.

Reply:
Got it, thanks!